### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Gegardus**
Implementing forms were created:

- a form for adding posts, updated the routes and controller to add data
- a form for adding comments, updated the routes and controller to add data
- a form for adding likes for comments.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    @post = Post.find(params[:post_id])
+    @comment.author_id = current_user.id
+    @comment.post_id = @post.id
+
+    if @comment.save
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: 'An error has occurred while creating the comment'
+    end
+  end
```
Comment (Cathella) [inline app/controllers/comments_controller.rb:17]: ✅ A comment is created on behalf of the currently logged in user (current_user) of the app 👍🏻 

Kindly refactor the create action to 

```ruby
def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end
```

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @like = Like.new(author_id: @user.id, post_id: @post.id)
+
+    if @like.save
+      @post = Post.find(params[:post_id])
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id),
+                  alert: 'An error has occurred while creating the like'
+    end
+  end
```
Comment (Cathella) [inline app/controllers/likes_controller.rb:18]: ✅ A user is able to add a like to posts 👍🏻 

Kindly refactor the create action to

```ruby
def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -8,4 +8,25 @@ def show
...
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    if @post.save
+      redirect_to user_path(id: @post.author_id)
+    else
+      render :new, alert: 'An error has occurred while creating the post'
+    end
+  end
```
Comment (Cathella) [inline app/controllers/posts_controller.rb:25]: ✅ A post is created on behalf of the currently logged in user (current_user) of the app 👍🏻 

Kindly refactor this to 
```ruby
def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end
```

File: app/views/comments/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<section>
+  <h1 class="users-title">Add comment</h1>  
+  <%= form_with model: @comment, url: user_post_comments_path, class: "card left" do |form| %>
+    <%= form.label :text %> <br>
+    <%= form.text_area :text , cols: 40, rows: 10 %> <br>
+    <%= form.submit "Submit", class: "btn" %>
+  <% end %>
```
Comment (Cathella) [inline app/views/comments/new.html.erb:7]: ✅ There is a form in the views/comments/new.html.erb file which accepts input for creating comments 👍🏻

File: app/views/posts/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<section>
+  <h1 class="users-title">Add a new post</h1>
+  <%= form_with model: @post, url: user_posts_path, class: "card left" do |form| %>
+    <%= form.label :title %> <br>
+    <%= form.text_field :title %> <br>
+    <%= form.label :text %> <br>
+    <%= form.text_area :text , cols: 40, rows: 10 %> <br>
+    <%= form.submit "Submit", class: "btn" %>
+  <% end %>
```
Comment (Cathella) [inline app/views/posts/new.html.erb:9]: ✅ There is a form in the views/posts/new.html.erb file which accepts input for creating posts 👍🏻

File: config/routes.rb [hunk 6]
```diff
@@ -3,7 +3,10 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-  resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create]
+      resources :likes, only: [:new, :create]
+    end
```
Comment (Cathella) [inline config/routes.rb:9]: ✅ There are routes in the `config/routes.rb` file for creating posts 👍🏻 
✅ There are routes in the config/routes.rb file for creating comments.

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @Gegardus 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💪🏻

✅ No linter errors.
✅ You used correct GitHub Flow.
✅ Professional README.
✅ Descriptive pull request.


Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gegardus** [comment]
Thank you @Cathella for your approval and suggestions. Have a great day!

**Pull request merged**